content/en/blog/2021/wasm-api-alpha/index.md

```diff
@@ -0,0 +1,107 @@
+
+First of all, some of the WASI (WebAssembly System Interface) system calls are supported now. For example, `clock_time_get` system call can be made from Wasm programs, which means that you can use `std::time::SystemTime::now()` in Rust or `time.Now().UnixNano()` in Go for Envoy Wasm extensions just like any other native platform. Another example is that `random_get` is supported by Envoy now, so the "crypto/rand" package is available in Go as a cryptographically secure random number generator. We are currently actively looking into file system support as we have seen the requests for reading and writing local files from Wasm programs running in Envoy.
+
+Next is the improvement in debuggability. Now the Envoy runtime emits the stack trace of your program when it causes runtime errors, for example, when null pointer exceptions occur in C++, or the panic function is called in Go or Rust. Previously Envoy did not say anything about the cause of errors, but now you can see the trace which can be used to debug your programs:
+
+{{< text plain >}}
+Function: proxy_on_request_headers failed: Uncaught RuntimeError: unreachable
+Proxy-Wasm plugin in-VM backtrace:
+  0:  0xdbd - runtime._panic
+  1:  0x103ab - main.anotherCalculation
+  2:  0x10399 - main.someCalculation
+  3:  0xea57 - main.myHeaderHandler
+  4:  0xea15 - proxy_on_request_headers
+{{< /text >}}
+
+The above is an example output of the stack trace for Go SDK based Wasm extensions. You might notice that you want file names and lines in the trace lines -- that is another huge future work and is currently an open issue since it has something to do with the DWARF format for WebAssembly and the Exception Handling proposal to WebAssembly specification.
+
+Also you can see strace equivalent logs emitted by Envoy. With Istio proxy’s component log level `wasm:trace`, we can observe all the system calls and Proxy-Wasm ABI calls that go across the boundary between Wasm virtual machines and Envoy. The following is an example log stream of such strace logs:
+
+{{< text plain >}}
+[host->vm] proxy_on_context_create(2, 1)
+[host<-vm] proxy_on_context_create return: void
+[host->vm] proxy_on_request_headers(2, 8, 1)
+[vm->host] wasi_snapshot_preview1.random_get(86928, 32)
+[vm<-host] wasi_snapshot_preview1.random_get return: 0
+[vm->host] env.proxy_log(2, 87776, 18)
+{{< /text >}}
+
+This is especially useful to debug Wasm program’s execution at runtime, for example, to verify it is not making any malicious system calls.
+
+The next one is about metrics. Wasm extensions have been able to define their own custom metrics and expose them in Envoy just like any other metric, but prior to Istio 1.12, all of these custom metrics are prefixed by `envoy_` Prometheus namespace and users were not be able to have their own namespaces. Now, you can choose whatever namespace you want, and your metrics are exposed in Envoy as-is without being prefixed by `envoy_`.
```

Comment:
Link to additional configuration needed to expose it from prom. Stats inclusion etc
@bianpengyuan

Reply:
hope this is what you meant :D https://github.com/istio/istio.io/pull/10530/commits/5b4eb8a97de1dbc405bd05d972d2a53f1188df13